gitenberg/config.py

```diff
@@ -23,82 +23,68 @@ class ConfigFile(object):
     for testing creation/destruction """
     # TODO emit warning if config doesn't exist
     # TODO make subcommand for creating config file
-    appname = 'gitberg'
+    APP_NAME = 'gitberg'
     file_name = 'config.yaml'
 
-    def __init__(self, appname=None):
-        if appname:
-            self.appname = appname
-        self.dir = appdirs.user_config_dir(self.appname)
-        self.exists_or_make()
-        self.parse()
+    def __init__(self):
+        self.app_dir = appdirs.user_config_dir(self.APP_NAME)
+        self._get_or_create_config()
+        self._load()
+
+    def __repr__(self):
+        return self._read()
 
     @property
     def file_path(self):
-        return os.path.join(self.dir, self.file_name)
-
-    def exists_or_make(self):
-        if not os.path.exists(self.dir):
-            os.makedirs(self.dir)
-        if not os.path.exists(self.file_path):
-            # FIXME: copy or template sample config file
-            with open(self.file_path, 'a'):
-                os.utime(self.file_path, None)
+        return os.path.join(self.app_dir, self.file_name)
 
     def write(self):
-        # FIXME: VVV
-        # self.check_self()
         with open(self.file_path, 'wb') as self.file:
             self.file.write(self.yaml)
         return True
 
-    @property
     def yaml(self):
-        return yaml.dump(data,
+        return yaml.dump(self.data,
                          default_flow_style=False)
 
-    def __repr__(self):
-        return self.read()
+    def _get_or_create_config(self):
+        if not os.path.exists(self.app_dir):
+            os.makedirs(self.app_dir)
+        if not os.path.exists(self.file_path):
+            # FIXME: copy or template sample config file
+            with open(self.file_path, 'a'):
+                os.utime(self.file_path, None)
 
-    def read(self):
+    def _load(self):
+        self.data = yaml.load(self._read())
+
+    def _read(self):
         with open(self.file_path) as _fp:
             return _fp.read()
 
-    def parse(self):
-        global data
```

Comment:
@eshellman Pulling a global out of the app, all at once, when everything is using it is certainly not good practice.  My intent is to go through the repo and add better test coverage before serious refactors.  I will:
+ restore the `global data`
+ add a separate way for modules to get a config without relying on data

Then I think it makes sense to modify this core object.